client/src/tasks_sidebar.ts

```diff
@@ -144,47 +124,45 @@ class DenoTaskProvider implements TaskProvider {
   }
 
   async provideTasks(): Promise<Task[]> {
-    const tasks: Task[] = [];
-
     const process = await getDenoCommandName();
-
-    // we retrieve config tasks from the language server, if the language server
-    // supports the capability
     const client = this.#extensionContext.client;
     const supportsConfigTasks = this.#extensionContext.serverCapabilities
       ?.experimental?.denoConfigTasks;
-    if (client && supportsConfigTasks) {
-      try {
-        const configTasks = await client.sendRequest(taskReq);
-        for (const { name, detail: command, sourceUri } of configTasks ?? []) {
-          const workspaceFolder = (workspace.workspaceFolders ?? []).find((f) =>
-            // NOTE: skipEncoding in Uri.toString(), read more at https://github.com/microsoft/vscode/commit/65cb3397673b922c1b6759d145a3a183feb3ee5d
-            sourceUri
-              .toLocaleLowerCase()
-              .startsWith(f.uri.toString(true).toLocaleLowerCase())
-          );
-          if (!workspaceFolder) {
-            continue;
-          }
-
-          tasks.push(
-            buildDenoConfigTask(
-              workspaceFolder,
-              process,
-              name,
-              command,
-              Uri.parse(sourceUri),
-            ),
-          );
+    if (!client || !supportsConfigTasks) {
+      return [];
+    }
+    const tasks = [];
+    try {
+      const configTasks = await client.sendRequest(taskReq);
+      for (const configTask of configTasks ?? []) {
+        const workspaceFolders = Array.from(
+          workspace.workspaceFolders ?? [],
+        );
+        workspaceFolders.reverse();
+        const workspaceFolder = workspaceFolders.find((f) =>
+          // NOTE: skipEncoding in Uri.toString(), read more at https://github.com/microsoft/vscode/commit/65cb3397673b922c1b6759d145a3a183feb3ee5d
```

Comment:
I realize this is previous code, but it might be good if we explained why `skipEncoding` here.